### PR TITLE
fix: refine validation rules as part of pre-1.0 work (group 6)

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit covers the following rules:
- ibm-major-version-in-path
- ibm-merge-patch-properties
- ibm-required-property-missing
- ibm-etag-header-missing

Signed-off-by: Phil Adams <phil_adams@us.ibm.com>

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Dependencies have been updated as needed
- [ ] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [ ] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [ ] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [ ] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [ ] Added tests for new rule (packages/ruleset/test/*.test.js)
- [ ] Added docs for new rule (docs/ibm-cloud-rules.md)

### CODE REVIEW

File: packages/ruleset/test/merge-patch-optional-properties.test.js [hunk 1]
```diff
@@ -2,12 +2,12 @@ const { mergePatchOptionalProperties } = require('../src/rules');
 const { makeCopy, rootDocument, testRule, severityCodes } = require('./utils');
 
 const rule = mergePatchOptionalProperties;
-const ruleId = 'ibm-merge-patch-optional-properties';
+const ruleId = 'ibm-merge-patch-properties';
```
Comment (pyrooka) [inline packages/ruleset/test/merge-patch-optional-properties.test.js:5]: I'm just wondering, why are the rule name and id different here? (`mergePatchOptionalProperties` vs `ibm-merge-patch-properties`)
I see there are a few cases where they are not the same, so I guess this is intentional.

**Review by pyrooka — APPROVED** [review]
LGTM!

File: packages/ruleset/test/merge-patch-optional-properties.test.js (hunk 1 above)
Comment (padamstx) [reply packages/ruleset/test/merge-patch-optional-properties.test.js]: done...

**Review by dpopp07 — APPROVED** [review]
👍

**Pull request merged**